modules/cdc-ext/src/main/java/org/apache/ignite/cdc/kafka/KafkaToIgniteCdcStreamerApplier.java

```diff
@@ -246,6 +253,9 @@ private boolean filterAndPossiblyUpdateMetadata(ConsumerRecord<Integer, byte[]>
         if (rec.key() == null && Arrays.equals(val, META_UPDATE_MARKER)) {
             metaUpdr.updateMetadata();
 
+            metrics.incrementMarkers();
+            metrics.decrementReceivedEvents();
```

Comment:
It looks strange when we first increment marker and later decrement it on some event types. Let's filter out meta records before incrementing events count.

Reply:
Refactored this part of the code. Moved increment for received events to `#desrialize()` method. We use it only for filtered events, which is the number of received events